datasets/pl/wanted/crawler.py

```diff
@@ -0,0 +1,160 @@
+
+
+def clean_and_split_citizenship(citizenship: str) -> str | List[str]:
+    """
+    Try to split the space-delimited citizenship field in various creative ways
+
+    Args:
+        citizenship: The string content of the citizenship field
+
+    Returns:
+        A string or a list of string containing the cleaned citizenship(s).
+
+    """
+    # Sometimes, we have another spelling of the country in brackets, like "USA (STANY ZJEDNOCZONE AMERYKI)"
+    alt_country_match = re.search(r"\((?P<alt_country>.+)\)", citizenship)
+    if alt_country_match:
+        cleaned_citizenship = re.sub(r" \(.+\)", "", citizenship)
+    else:
+        cleaned_citizenship = citizenship
+
+    if countrynames.to_code(cleaned_citizenship, fuzzy=True):
+        return cleaned_citizenship
+
+    if " " in cleaned_citizenship:
+        split_citizenships = cleaned_citizenship.split(" ")
+        # Somtimes a person is "NIEUSTALONE POLSKA" aka unknown polish
+        split_citizenships = [
+            c for c in split_citizenships if c not in NULL_CITIZENSHIPS
+        ]
+
+        # Multiple citizenships are space-delimited (yay), see if space-splitting them works
+        if all([countrynames.to_code(x) for x in split_citizenships]):
+            return split_citizenships
+        else:
+            # Sometimes, the citizenship looks like this "WIELKA BRYTANIA IZRAEL"
+            # Covering this special case here (just one split) is easy and cheap
+            for i in range(1, len(split_citizenships) + 1):
+                citizenship_1 = " ".join(split_citizenships[:i])
+                citizenship_2 = " ".join(split_citizenships[i:])
+                if countrynames.to_code(
+                    citizenship_1, fuzzy=True
+                ) and countrynames.to_code(citizenship_2, fuzzy=True):
+                    return [citizenship_1, citizenship_2]
+
+    # Sometimes, there is a full spelling of the country in parens
+    if alt_country_match:
+        alt_country = alt_country_match.groupdict().get("alt_country")
+        if countrynames.to_code(alt_country, fuzzy=True):
+            return alt_country
+
+    # If nothing worked, we just return the original value
+    return citizenship
+
+
+def crawl_person(context: Context, url: str):
+    context.log.debug(f"Crawling person page {url}")
+
+    person = context.make("Person")
+    # TODO(Leon Handreke): Do we use the r12323456 in the URL that looks like a record ID?
+
+    # There is an ID prefixed by r in the URL, which seems to refer to the record. The name component in the URL
+    # can be changed at will.
+    # URLs look like this: https://poszukiwani.policja.gov.pl/pos/form/r922391307881,OCHNIO-KRZYSZTOF.html
+    url_match = re.search(r"/pos/form/(?P<rid>r\d+),.+.html", url)
+    assert (
+        url_match and "rid" in url_match.groupdict()
+    ), "Could not extract r-ID from malformed person URL"
+    rid = url_match.groupdict()["rid"]
+    person.id = context.make_slug(rid)
```

Comment:
it's tempting, but it could be an artifact of their software system rather than some operational consistent identifier. If it's the former, I worry a bit that it's not stable enough, and I'd feel more comfortable with context.make_id(full_name, birth_date) and maybe chucking place of birth in there too, especially if birth date isn't super consistently available.

Reply:
Done.